### PR TITLE
Add support for running gohan tests with mysql (MYSQL_TEST flag)

### PR DESCRIPTION
**Pull request by przemyslaw-dobrowolski-cl**

### CODE REVIEW

File: extension/framework/goplugin/runner/runner.go [hunk 1]
```diff
@@ -180,9 +182,19 @@ func (testRunner *TestRunner) runSingle(t ginkgo.GinkgoTestingT, reporter *Repor
...
 		// db
-		dbFileName := dbBaseFileName + "_" + uuid.NewV4().String()
-		dbConnString := fmt.Sprintf("file:%s?mode=memory&cache=shared", dbFileName)
-		db, err := gohan_db.ConnectDB("sqlite3", dbConnString, gohan_db.DefaultMaxOpenConn, options.Default())
+		dbType := ""
+		dbConn := ""
+
+		if os.Getenv("MYSQL_TEST") == "true" {
+			dbType = "mysql"
+			dbConn = "root@/gohan_test"
```
Comment (zimnx) [inline extension/framework/goplugin/runner/runner.go]: what about taking env variable MYSQL_DSN with this as default?

Comment (przemyslaw-dobrowolski-cl) [reply extension/framework/goplugin/runner/runner.go]: This variable is never used in gohan. What kind of usage you have in mind?

Comment (zimnx) [reply extension/framework/goplugin/runner/runner.go]: I mean that I have different credentials to my local mysql server. I would need to recompile gohan if I would like to run tests. Getting DSN string from env variable is much better solution.

Comment (przemyslaw-dobrowolski-cl) [reply extension/framework/goplugin/runner/runner.go]: done, added optional DSN parameter

File: extension/framework/goplugin/runner/runner.go [hunk 2]
```diff
@@ -180,9 +182,23 @@ func (testRunner *TestRunner) runSingle(t ginkgo.GinkgoTestingT, reporter *Repor
...
-		db, err := gohan_db.ConnectDB("sqlite3", dbConnString, gohan_db.DefaultMaxOpenConn, options.Default())
+		dbType := ""
+		dbConn := ""
+
+		if os.Getenv("MYSQL_TEST") == "true" {
+			dbType = "mysql"
+			if dsn, ok := os.LookupEnv("MYSQL_DSN"); ok {
+				dbConn = dsn
+			} else {
+				dbConn = "root@/gohan_test"
```
Comment (morrisson) [inline extension/framework/goplugin/runner/runner.go]: Is hardcode ok?

File: extension/framework/goplugin/runner/runner.go [hunk 3]
```diff
@@ -22,6 +22,8 @@ import (
 	"regexp"
 	"testing"
 
+	"os"
```
Comment (p-kozlowski) [inline extension/framework/goplugin/runner/runner.go]: go fmt

File: extension/framework/goplugin/runner/runner.go (hunk 2 above)
Comment (przemyslaw-dobrowolski-cl) [reply extension/framework/goplugin/runner/runner.go]: It's not the best solution here. There were a dozen of similar places in code where connection was hardcoded. I added a commit which contains a common file for local database connection and clears Connect and InitSchema functions naming.

**Pull request closed without merging**